backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresExtension.java

```diff
@@ -180,12 +181,33 @@ public ConnectionFactory getConnectionFactory() {
     }
 
     private void initTablesAndIndexes() {
-        postgresTableManager = new PostgresTableManager(postgresExecutor, postgresModule, postgresConfiguration.rowLevelSecurityEnabled());
+        PostgresTableManager postgresTableManager = new PostgresTableManager(postgresExecutor, postgresModule, postgresConfiguration.rowLevelSecurityEnabled());
         postgresTableManager.initializeTables().block();
         postgresTableManager.initializeTableIndexes().block();
     }
 
     private void resetSchema() {
-        postgresTableManager.truncate().block();
+        dropTables(listAllTables());
+    }
+
+    private void dropTables(List<String> tables) {
```

Comment:
Why do we need it?
We create some tables that are not by `postgresTableManager`?

Reply:
> We create some tables that are not by postgresTableManager?

Yes. `PostgresTableManagerTest` creates some custom tables, and JPA tables also not managed by PostgresTableManager now. Having the extension handles cleaning those tables would be good IMO.

BTW the rationale behind the change (before it was reset the whole schema) to now just drop all tables: is because reset schema would remove the HSTORE extension, then the connection needs to be re-inited to set the hstore decoding again (otherwise hstore test fails). Today we share the same PostgreSQL connection for all tests, so I switched to only DROP tables to avoid removing the HSTORE extension and reconnecting the connection (leading the the tests run time longer).